### ISSUE TITLE
🐛 [Bug] Undefined symbol when importing `torch_tensorrt`

### ISSUE THREAD

**Issue by laclouis5** [pytorch/TensorRT#3031]
##  Bug Description

Importing `torch_tensorrt` fails with the following error:

```shell
Traceback (most recent call last):
  File "/home/louis/Documents/inference_opti_onnx_tensorrt/export.py", line 27, in <module>
    import torch_tensorrt as trt
  File "/home/louis/Documents/inference_opti_onnx_tensorrt/.venv/lib/python3.11/site-packages/torch_tensorrt/__init__.py", line 107, in <module>
    _register_with_torch()
  File "/home/louis/Documents/inference_opti_onnx_tensorrt/.venv/lib/python3.11/site-packages/torch_tensorrt/__init__.py", line 101, in _register_with_torch
    torch.ops.load_library(trtorch_dir + linked_file)
  File "/home/louis/Documents/inference_opti_onnx_tensorrt/.venv/lib/python3.11/site-packages/torch/_ops.py", line 1032, in load_library
    ctypes.CDLL(path)
  File "/home/louis/.pyenv/versions/3.11.4/lib/python3.11/ctypes/__init__.py", line 376, in __init__
    self._handle = _dlopen(self._name, mode)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
OSError: /home/louis/Documents/inference_opti_onnx_tensorrt/.venv/lib/python3.11/site-packages/torch_tensorrt/lib/libtorchtrt.so: undefined symbol: _ZNK3c105Error4whatEv
```

## To Reproduce

I followed the official installation tutorial:

```shell
python3.11 -m venv .venv
source .venv/bin/activate
pip install torch-tensorrt
```

I also tried the method from the Pytorch blog, which is slightly different:

```shell
python3.11 -m venv .venv
source .venv/bin/activate
pip install torch torch-tensorrt tensorrt
```

They both fails with the same import error.

However, using the CUDA 11.8 wheels seems to work fine:

```shell
python3.11 -m venv .venv
source .venv/bin/activate
pip install torch torch-tensorrt tensorrt --extra-index-url https://download.pytorch.org/whl/cu118
```

## Expected behavior

The import should work properly.

## Environment

> Build information about Torch-TensorRT can be found by turning on debug messages

 - Torch-TensorRT Version (e.g. 1.0.0): 10
 - PyTorch Version (e.g. 1.0): 2.3
 - CPU Architecture: x86_64
 - OS (e.g., Linux): Ubuntu
 - How you installed PyTorch (`conda`, `pip`, `libtorch`, source): pip
 - Build command you used (if compiling from source): See above
 - Are you using local sources or building from archives: No
 - Python version: 3.11
 - CUDA version: 12.1
 - GPU models and configuration: RTX 3070

**Comment by lanluo-nvidia** [comment]
Yes, I can confirm I am able to reproduce the same issue as described.
I am looking into this issue now.

**Comment by lanluo-nvidia** [comment]
I have verified that It was fixed in our 2.4 release, it should be available on Monday.
